src/core/server/http/base_path_proxy_server.test.ts

```diff
@@ -0,0 +1,1148 @@
+    registerRouter(router);
+
+    await server.start();
+
+    await proxySupertest
+      .delete(`${basePath}/foo/3`)
+      .expect(200)
+      .then((res) => {
+        expect(res.body).toEqual({ key: 3 });
+      });
+  });
+
+  describe('with `basepath: /bar` and `rewriteBasePath: false`', () => {
+    let configWithBasePath: HttpConfig;
+
+    beforeEach(async () => {
+      configWithBasePath = {
+        ...config,
+        basePath: '/bar',
+        rewriteBasePath: false,
+      } as HttpConfig;
+
+      const router = new Router(`${basePath}/`, logger, enhanceWithContext);
+      router.get({ path: '/', validate: false }, (_, __, res) => res.ok({ body: 'value:/' }));
+      router.get({ path: '/foo', validate: false }, (_, __, res) => res.ok({ body: 'value:/foo' }));
+
+      const { registerRouter } = await server.setup(configWithBasePath);
+      registerRouter(router);
+
+      await server.start();
+    });
+
+    test('/bar => 404', async () => {
+      await proxySupertest.get(`${basePath}/bar`).expect(404);
+    });
+
+    test('/bar/ => 404', async () => {
+      await proxySupertest.get(`${basePath}/bar/`).expect(404);
+    });
+
+    test('/bar/foo => 404', async () => {
+      await proxySupertest.get(`${basePath}/bar/foo`).expect(404);
+    });
+
+    test('/ => /', async () => {
+      await proxySupertest
+        .get(`${basePath}/`)
+        .expect(200)
+        .then((res) => {
+          expect(res.text).toBe('value:/');
+        });
+    });
+
+    test('/foo => /foo', async () => {
+      await proxySupertest
+        .get(`${basePath}/foo`)
+        .expect(200)
+        .then((res) => {
+          expect(res.text).toBe('value:/foo');
+        });
+    });
+  });
+
+  test('with defined `redirectHttpFromPort`', async () => {
+    const router = new Router(`${basePath}/`, logger, enhanceWithContext);
+    router.get({ path: '/', validate: false }, (_, __, res) => res.ok({ body: 'value:/' }));
+
+    const { registerRouter } = await server.setup(configWithSSL);
+    registerRouter(router);
+
+    await server.start();
+  });
+
+  test('allows attaching metadata to attach meta-data tag strings to a route', async () => {
+    const tags = ['my:tag'];
+    const { registerRouter } = await server.setup(config);
+
+    const router = new Router(basePath, logger, enhanceWithContext);
+    router.get({ path: '/with-tags', validate: false, options: { tags } }, (_, req, res) =>
+      res.ok({ body: { tags: req.route.options.tags } })
+    );
+    router.get({ path: '/without-tags', validate: false }, (_, req, res) =>
+      res.ok({ body: { tags: req.route.options.tags } })
+    );
+    registerRouter(router);
+
+    await server.start();
+    await proxySupertest.get(`${basePath}/with-tags`).expect(200, { tags });
+
+    await proxySupertest.get(`${basePath}/without-tags`).expect(200, { tags: [] });
+  });
+
+  test('exposes route details of incoming request to a route handler', async () => {
```

Comment:
all request handler tests below can be deleted as proxy doesn't contain any logic for them

Reply:
Thanks, will do!

Reply:
For the timeout tests, so you know were showing some differing behavior where I had to add these parts to the test:

```ts
            timeout: {
              payload: 300000,
              idleSocket: 1000, // This is an extra option added by the proxy
            },
```

I don't know if you want the `idleSocket` to not be there compared to the regular http request/response routes when no base proxy is in front of them but I did have to add that for those tests to pass. I'm not removing those but am going to remove these tests for `requests`:

* exposes route details of incoming request to a route handler
* should reject the request because the Content-Type in the request is not valid
* should reject the request because the payload is too large
* should not parse the content in the request

I was taking the approach that as close to a 1 to 1 match with testing even if we didn't have custom logic was better to see if I could expose/test different behaviors of h2o2 since even though we don't have custom logic they can/do when we upgrade compared to how we need/want h2o2 to behave.

But I do think that practically tests are better to eventually push down into h2o2 library from their authors than have them at our layer as that just adds bulk to our unit tests. But having some safety checks for upgrades (like these show off) will be good enough even with those 4 removed.